client/src/modules/stock/lots/registry.service.js

```diff
@@ -0,0 +1,139 @@
+      displayName : 'TABLE.COLUMNS.INVENTORY_GROUP',
+      headerCellFilter : 'translate',
+    }, {
+      field : 'label',
+      displayName : 'STOCK.LOT',
+      headerCellFilter : 'translate',
+    }, {
+      field : 'quantity',
+      displayName : 'STOCK.QUANTITY',
+      headerCellFilter : 'translate',
+    }, {
+      field : 'unit_cost',
+      displayName : 'STOCK.UNIT_COST',
+      headerCellFilter : 'translate',
+      type : 'number',
+      cellFilter : 'currency: '.concat(Session.enterprise.currency_id),
+    }, {
+      field : 'unit_type',
+      width : 75,
+      displayName : 'TABLE.COLUMNS.UNIT',
+      headerCellFilter : 'translate',
+      cellTemplate : 'modules/stock/inventories/templates/unit.tmpl.html',
+    }, {
+      field : 'entry_date',
+      displayName : 'STOCK.ENTRY_DATE',
+      headerCellFilter : 'translate',
+      cellFilter : 'date',
+    }, {
+      field : 'expiration_date',
+      displayName : 'STOCK.EXPIRATION_DATE',
+      headerCellFilter : 'translate',
+      cellFilter : 'date',
+    }, {
+      field : 'delay_expiration',
+      displayName : 'STOCK.EXPIRATION',
+      headerCellFilter : 'translate',
+    }, {
+      field : 'avg_consumption',
+      displayName : 'STOCK.CMM',
+      headerCellFilter : 'translate',
+      type : 'number',
+    }, {
+      field : 'S_MONTH',
+      displayName : 'STOCK.MSD',
+      headerCellFilter : 'translate',
+      type : 'number',
+    }, {
+      field : 'lifetime',
+      displayName : 'STOCK.LIFETIME',
+      headerCellFilter : 'translate',
+      cellTemplate     : 'modules/stock/lots/templates/lifetime.cell.html',
+      type : 'number',
+      sort : {
+        direction : uiGridConstants.ASC,
+        priority : 2,
+      },
+    }, {
+      field : 'S_LOT_LIFETIME',
+      displayName : 'STOCK.LOT_LIFETIME',
+      headerCellFilter : 'translate',
+      cellTemplate     : 'modules/stock/lots/templates/lot_lifetime.cell.html',
+      type : 'number',
+    }, {
+      field : 'S_RISK',
+      displayName : 'STOCK.RISK',
+      headerCellFilter : 'translate',
+      cellTemplate     : 'modules/stock/lots/templates/risk.cell.html',
+      type : 'number',
+      sort : {
+        direction : uiGridConstants.DESC,
+        priority : 3,
+      },
+    }, {
+      field : 'S_RISK_QUANTITY',
+      displayName : 'STOCK.RISK_QUANTITY',
+      headerCellFilter : 'translate',
+      cellTemplate     : 'modules/stock/lots/templates/risk_quantity.cell.html',
+      type : 'number',
+    }, {
+      field : 'action',
+      displayName : '',
+      enableFiltering : false,
+      enableSorting : false,
+      cellTemplate : 'modules/stock/lots/templates/action.cell.html',
+    }];
+
+  service.gridFooterTemplate = `
+    <div>
+      <b>{{ grid.appScope.countGridRows() }}</b>
+      <span translate>TABLE.AGGREGATES.ROWS</span>
+    </div>
+  `;
+
+  service.orderByDepot = (rowA, rowB) => {
+    return rowA.depot_text > rowB.depot_text ? 1 : -1;
```

Comment:
Remember that strings without ASCII characters should be compared using [`String.prototype.localeCompare()`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/sort#Sorting_non-ASCII_characters).